python/src/lib.rs

```diff
@@ -1098,6 +1100,13 @@ fn save_mode_from_str(value: &str) -> PyResult<SaveMode> {
     }
 }
 
+fn partition_strategy_from_str(value: &str) -> PyResult<PartitionStrategy> {
+    match value {
+        "hive" => Ok(PartitionStrategy::Hive),
+        _ => Err(PyValueError::new_err("Invalid partition strategy provided")),
+    }
+}
```

Comment:
Would it make sense to have this as a `FromStr` implementation directly on the `PartitionStrategy` enum?

Reply:
Yes, definitely should be a small change. 

Then I guess we can also do this on the savemode enum

Reply:
@roeap I have added it on SaveMode and PartitionStategy